crates/node/src/txvalidation/event.rs

```diff
@@ -0,0 +1,185 @@
+use crate::txvalidation::download_manager;
+use crate::txvalidation::EventProcessError;
+use crate::types::{
+    transaction::{TransactionError, TxReceive, TxValdiated},
+    Transaction,
+};
+use crate::Mempool;
+use futures::future::join_all;
+use futures_util::TryFutureExt;
+use gevulot_node::types::transaction::AclWhitelist;
+use std::fmt::Debug;
+use std::net::SocketAddr;
+use std::path::Path;
+use tokio::sync::mpsc::UnboundedSender;
+
+//event type.
+#[derive(Debug, Clone)]
+pub struct RcvTx;
+
+#[derive(Debug, Clone)]
+pub struct DownloadTx;
+
+#[derive(Debug, Clone)]
+pub struct NewTx;
+
+#[derive(Debug, Clone)]
+pub struct PropagateTx;
+
+//Event processing depends on the marker type.
+#[derive(Debug, Clone)]
+pub struct EventTx<T: Debug> {
```

Comment:
```suggestion
pub struct TxEvent<T: Debug> {
```
Is this event transaction or transaction event?

Reply:
I think TxEvent event is better. I change it.